refactor/all: impl outcome type, remove panics

oetyng opened the pull request:
- Outcome is a ternary result; success, no change or error.
- Tests have been updated to return Result.
- Panics have been replaced with Error.
- NB: Unfortunately the Outcome type has led to unwraps in the tests.
To be solved.

Yoga07 commented on src/actor.rs:
```diff
@@ -122,7 +123,7 @@ impl<V: ReplicaValidator> Actor<V> {
-    pub fn transfer(&self, amount: Money, to: AccountId) -> Result<TransferInitiated> {
+    pub fn transfer(&self, amount: Money, to: AccountId) -> Outcome<TransferInitiated> {
         if to == self.id {
             return Err(Error::from("Sender and recipient are the same"));
```
Can we use `Outcome::rejected()` here - just to keep all the returns inline.

oetyng replied:
Hm, I trierd it now in the IDE, looks messier IMO?

oetyng replied:
Btw, how do you mean they would all be inline?

Yoga07 replied:
'In line', typo :) Meaning keeping the returns consistent by using `Outcome`

oetyng replied:
I see though that it is used above..
I think I will fix this up for next PR.

Yoga07 commented on src/replica.rs:
```diff
@@ -120,8 +120,8 @@ impl Replica {
     /// Get the replica's PK set
-    pub fn replicas_pk_set(&self) -> Result<PublicKeySet> {
-        Ok(self.peer_replicas.clone())
+    pub fn replicas_pk_set(&self) -> Option<PublicKeySet> {
```
Why are they Option-ized when we return them directly?

oetyng replied:
It had a `Resul`t, so that was just to follow the api convention (queries return `Option`, cmds return `Result`).
Then of course, if an Actor _cannot_ be instantiated without replica pks (which is the case now), there is no need for an Option. But, I saw that as a non important aspect, and decided to keep the api consistent.

oetyng committed:
refactor/all: impl outcome type, remove panics
- Outcome is a ternary result; success, no change or error.
- Tests have been updated to return Result.
- Panics have been replaced with Error.
- NB: Unfortunately the Outcome type has led to unwraps in the tests.
To be solved.
